### PR TITLE
[front] feat: add an already suggested cache to the Auto button

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related issues** #1743

**this other PR targets the current PR** https://github.com/tournesol-app/tournesol/pull/1760 (merged), #1782 (merged)

---

### Description

This PR adds a quick and not-so-dirty mechanism that will prevent the same video to be suggested several times by the Auto button.

The suggested videos are cached per poll. The cache is flushed when its size reaches the configured limit of 60. It's then possible for a user to see two times the same video after 61 clicks on the Auto button.

When the /suggestions/ API will be implemented, if this cache is still useful and if it becomes more complex, we may want to move it in the Redux store. 

**to-do**
- [x] ensure auto-suggestion are added to the `ALREADY_SUGGESTED` list
- [x] handle different polls

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
  - if not, delete the related issues section
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [x] The tests pass and have been updated if relevant
- [x] The code quality check pass

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

### CODE REVIEW

**Comment by GresilleSiffle** [comment]
I created a new branch that targets this one to isolate the refactor from the new feature. 

https://github.com/tournesol-app/tournesol/pull/1782

File: frontend/src/utils/video.ts [hunk 1]
```diff
@@ -124,59 +133,83 @@ export async function getVideoFromPreviousComparisons(
...
   ];
-  const comparisonVideoId = retryRandomPick(
-    5,
-    otherVideo,
-    currentVideo,
-    comparisonVideoList
+
+  let newSuggestions = comparisonVideoList.filter(
+    (uid, index) => comparisonVideoList.indexOf(uid) === index
   );
```
Comment (amatissart) [inline frontend/src/utils/video.ts:151]: Instead of deduplicating comparisons, maybe we can fetch the last compared entities using the ratings API?

Comment (GresilleSiffle) [reply frontend/src/utils/video.ts]: Fixed https://github.com/tournesol-app/tournesol/pull/1758/commits/dc43c042ab8b794d1cecc70547a3f8b0c0eb28e8

**Review by amatissart — APPROVED** [review]

File: frontend/src/features/rateLater/alreadySuggested.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const MAX_LENGTH = 60;
+
+export const ALREADY_SUGGESTED: { [key: string]: string[] } = {};
+
+const initSuggested = (poll: string) => (ALREADY_SUGGESTED[poll] = []);
+
+export const clearSuggested = (poll: string) =>
+  (ALREADY_SUGGESTED[poll].length = 0);
```
[the remark this answers is not in the document]
Comment (GresilleSiffle) [reply frontend/src/features/rateLater/alreadySuggested.ts]: Fixed by https://github.com/tournesol-app/tournesol/pull/1758/commits/940f9bbd3b646d9e5db28c00d78143dfbd42792d

**Review by amatissart — APPROVED** [review]

**Pull request merged**